code/modules/mob/living/carbon/superior_animal/roach/types/support.dm

```diff
@@ -41,6 +41,9 @@
 
 	gas_sac.add_reagent("blattedin", 1)
 
+	if(!target_mob)
+		return
+
```

Comment:
I wonder if the proc life of cockroaches and of all superior animals continues to run after death, and that is the biggest problem. I ask because in this code I do not see that verification.

Reply:
From what I can tell through the power of var editing, `Life()` does indeed stop ticking once the roach has been `Death()`'d, so I suppose this won't need a "if alive" check.

Reply:
don't worry, I studied the life code better and everything seems fine.